### PR TITLE
disabling GUI buttons of forced External Storage

### PR DESCRIPTION
**Pull request by paulr34**
This PR addresses both issues:
https://github.com/project-chip/zap/issues/1086, 
https://github.com/project-chip/zap/issues/1145

If attribute storage policy is attributeAccessInterface then the storage and the default field are not editable. It also sets default value to null.

The default field is still there but it is disabled.

### CODE REVIEW

**Comment by brdandu** [comment]
Seeing a lot of changes to the matter code gen. Is this expected? See all .matter files with diffs.

**Review by brdandu — CHANGES_REQUESTED** [review]
Add tests to show that when external storage option is used then only Matter is affected and not Zigbee

**Comment by paulr34** [comment]
> Seeing a lot of changes to the matter code gen. Is this expected? See all .matter files with diffs.

Yes, they look right. For example, in Access Control we see 
-    callback attribute clusterRevision default = 1;
+    callback attribute clusterRevision;

the default for this external attribute is set to null

**Comment by paulr34** [comment]
> Add tests to show that when external storage option is used then only Matter is affected and not Zigbee

Zigbee doesn't have attributeAccessInterface 

Good catch, I made this commit to limit it to Matter data https://github.com/project-chip/zap/pull/1181/commits/3bc7a3a6116922852290de2843ac02aba660fbe0

As long as Zigbee does not have attributeAccessInterface in their zcl.json then it won't effect it but it would definitely be a good idea to add a cypress test

File: src-electron/db/query-config.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ const dbMapping = require('./db-mapping.js')
 const queryPackage = require('./query-package.js')
 const dbEnum = require('../../src-shared/db-enum.js')
 const queryZcl = require('./query-zcl.js')
-const queryUpgrade = require('../upgrade/upgrade.js')
+const queryUpgrade = require('../matter/matter.js')
```
Comment (tecimovic) [inline src-electron/db/query-config.js:28]: As a matter of general rules: we should NOT use the `.js` suffix in require calls.
That allows you to migrate the specific file to a different language (like TS) without having to modify these requires.

File: src-electron/rest/user-data.js [hunk 2]
```diff
@@ -108,7 +109,8 @@ function httpGetPackageNotifications(db) {
 function httpGetPackageNotificationsByPackageId(db) {
   return async (request, response) => {
     let packageId = request.params.packageId
-    let notifications = await queryPackageNotification.getNotificationByPackageId(db, packageId)
+    let notifications =
+      await queryPackageNotification.getNotificationByPackageId(db, packageId)
```
Comment (tecimovic) [inline src-electron/rest/user-data.js:113]: Why do we have these formatting diffs? Do we not all use the same formatter?

**Review by tecimovic — APPROVED** [review]

File: src-electron/rest/user-data.js (hunk 2 above)
Comment (paulr34) [reply src-electron/rest/user-data.js:113]: not sure

**Pull request merged**